### PR TITLE
cvar: check error returns for MPIR_T_env_init

### PR DESCRIPTION
**Pull request by hzhou**
## Pull Request Description

We control a large portion of features using CVAR. It is a common mistake to supply the wrong CVAR values resulting in the desired feature option being silently dropped.

This PR enables the checking of invalid CVAR enum values and raise errors in the init routines.

## Author Checklist
* [ ] Reference appropriate issues (with "Fixes" or "See" as appropriate)
* [ ] Passes tests (included warning check)
* [x] Confirm whitespace/style checkers are happy (or has a good reason for being bad)
* [x] Commits are self-contained and do not do two things at once
* [ ] Remove xfail from the test suite when fixing a test
* [x] Commit message is of the form: `module: short description` and follows [good practice](https://chris.beams.io/posts/git-commit/)
* [ ] Add comments such that someone without knowledge of the code could understand
* [ ] Add Devel Docs in the `doc/` directory for any new code design

### CODE REVIEW

**Comment by hzhou** [comment]
test:jenkins/ch3/most
test:jenkins/ch4/most

**Comment by hzhou** [comment]
test:jenkins/ch3/most
test:jenkins/ch4/most

File: src/mpi/init/init.c [hunk 1]
```diff
@@ -153,7 +153,9 @@ int MPI_Init(int *argc, char ***argv)
     MPIR_ThreadInfo.isThreaded = 0;
 #endif /* MPICH_IS_THREADED */
 
-    MPIR_T_env_init();
+    mpi_errno = MPIR_T_env_init();
+    if (mpi_errno != MPI_SUCCESS)
```
Comment (raffenet) [inline src/mpi/init/init.c]: Is this is place to use the new `MPIR_ERR_CHECK` macro?

Comment (hzhou) [reply src/mpi/init/init.c]: `MPIR_ERR_CHECK` will call `MPIR_ERR_POP`, which will insert an entry into the error chain. In the current code there is such error insertion at `fn_fail` already, therefore, using `MPIR_ERR_POP` or `MPIR_ERR_CHECK` would be redundant.

Comment (raffenet) [reply src/mpi/init/init.c]: OK, I got it.

File: maint/extractcvars.in [hunk 2]
```diff
@@ -419,14 +419,10 @@ EOT
...
-        # FIXME: There is no failure mode designed for the entire MPIR_T_..._init currently
-        #        MPIR_T_cvar_env_init and MPIR_T_env_init et el returns void
-        #        I don't see why not, but the change is another PR.
-        #        Once we checks the error code, uncomment the following.
-        # print OUTPUT_C $sp, "else {\n";
-        # print OUTPUT_C $sp, "    mpi_errno = MPIR_Err_create_code(MPI_SUCCESS, MPIR_ERR_FATAL,__func__,__LINE__,MPI_ERR_ARG, \"**cvar_val\", \"**cvar_val %s %s\", \"$var_name\", tmp_str);\n";
-        # print OUTPUT_C $sp, "    goto fn_fail;\n";
-        # print OUTPUT_C $sp, "}\n";
+        print OUTPUT_C $sp, "else {\n";
+        print OUTPUT_C $sp, "    mpi_errno = MPIR_Err_create_code(MPI_SUCCESS, MPIR_ERR_RECOVERABLE, __func__, __LINE__,MPI_ERR_OTHER, \"**cvar_val\", \"**cvar_val %s %s\", \"$var_name\", tmp_str);\n";
```
Comment (wesbland) [inline maint/extractcvars.in]: Should this be `MPI_ERR_OTHER` or should it stay as `MPI_ERR_ARG`. The later is more descriptive

Comment (hzhou) [reply maint/extractcvars.in]: `MPI_ERR_ARG` will say "Invalid argument". I find it mysterious as no "command line argument" nor "function argument" were supplied. Hopefully, the actual message "**cvar_val" is sufficiently clear.

Comment (wesbland) [reply maint/extractcvars.in]: That's fine then.

**Pull request merged**